backend/dataall/modules/s3_datasets/cdk/dataset_stack.py

```diff
@@ -437,20 +449,68 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
+            alias=f'crwlr_log_enc_key_{dataset.GlueCrawlerName}',
+            enable_key_rotation=True,
+            admins=[
+                iam.ArnPrincipal(env.CDKRoleArn),
+            ],
+        )
+
+        glue_sec_conf_enc_key.add_to_resource_policy(
+            iam.PolicyStatement(
+                sid='EnableCrawlerIAMRoleKeyUsage',
+                resources=['*'],
+                effect=iam.Effect.ALLOW,
+                principals=[dataset_admin_role],
+                actions=['kms:Encrypt', 'kms:Decrypt', 'kms:ReEncrypt*', 'kms:GenerateDataKey*', 'kms:DescribeKey'],
+            )
+        )
+
+        glue_sec_conf_enc_key.grant_encrypt_decrypt(iam.ServicePrincipal('logs.amazonaws.com'))
+        glue_sec_conf_enc_key.grant_encrypt_decrypt(iam.ServicePrincipal('glue.amazonaws.com'))
+        glue_sec_conf_enc_key.grant_encrypt_decrypt(iam.ServicePrincipal('s3.amazonaws.com'))
+
+        glue_crawler_security_config = glue.CfnSecurityConfiguration(
+            self,
+            f'crwlr_sec_config-{dataset.GlueCrawlerName}',
+            encryption_configuration=glue.CfnSecurityConfiguration.EncryptionConfigurationProperty(
+                cloud_watch_encryption=glue.CfnSecurityConfiguration.CloudWatchEncryptionProperty(
+                    cloud_watch_encryption_mode='SSE-KMS', kms_key_arn=glue_sec_conf_enc_key.key_arn
+                ),
+                job_bookmarks_encryption=glue.CfnSecurityConfiguration.JobBookmarksEncryptionProperty(
+                    job_bookmarks_encryption_mode='CSE-KMS', kms_key_arn=glue_sec_conf_enc_key.key_arn
+                ),
+                s3_encryptions=[
+                    glue.CfnSecurityConfiguration.S3EncryptionProperty(
+                        s3_encryption_mode='SSE-KMS', kms_key_arn=glue_sec_conf_enc_key.key_arn
+                    )
+                ],
+            ),
+            name=f'crwlr_sec_config-{dataset.GlueCrawlerName}',
+        )
+
         # Support resources: GlueCrawler for the dataset, Profiling Job and Trigger
         crawler = glue.CfnCrawler(
             self,
             dataset.GlueCrawlerName,
             description=f'datall Glue Crawler for S3 Bucket {dataset.S3BucketName}',
             name=dataset.GlueCrawlerName,
             database_name=dataset.GlueDatabaseName,
+            crawler_security_configuration=glue_crawler_security_config.name,
             schedule={'scheduleExpression': f'{dataset.GlueCrawlerSchedule}'} if dataset.GlueCrawlerSchedule else None,
             role=dataset_admin_role.role_arn,
             targets=CfnCrawler.TargetsProperty(
                 s3_targets=[CfnCrawler.S3TargetProperty(path=f's3://{dataset.S3BucketName}')]
             ),
         )
         crawler.node.add_dependency(dataset_bucket)
+        crawler.node.add_dependency(dataset_admin_policy)
+        crawler.node.add_dependency(dataset_admin_role)
+        crawler.node.add_dependency(glue_crawler_security_config)
```

Comment:
qq - do we need to add these `add_dependency(...)` statements? I would imagine there would be an implicit dependency since we are passing `crawler_security_configuration=glue_crawler_security_config.name,` as a parameter for the `CfnCrawler` but could be wrong

Reply:
this was required @noah-paige . For some reason, during deployment, it was looking for permissions on the policy and without the dependency, the role was getting created first without the custom policy and crawler creation was failing all the time.

Reply:
okay good data point to know - makes sense!